table-api/src/main/java/io/deephaven/api/TableOperations.java

```diff
@@ -0,0 +1,116 @@
+package io.deephaven.api;
+
+import io.deephaven.api.agg.Aggregation;
+import java.util.Collection;
+
+/**
+ * Table operations is a user-accessible api for modifying tables or building up table operations.
+ *
+ * @param <TOPS> the table operations type
+ * @param <TABLE> the table type
+ */
+public interface TableOperations<TOPS extends TableOperations<TOPS, TABLE>, TABLE> {
```

Comment:
We're missing a lot of ops, including a few very important joins. How far do we want to go in this PR?

Reply:
I've added leftJoin, aj, raj joins; and snapshot. I think it's useful, even though not complete. I'm hoping we can plan to merge shortly as-is, pulling over addition methods as necessary.